etor/src/main/java/gov/hhs/cdc/trustedintermediary/external/database/PostgresDao.java

```diff
@@ -128,6 +128,62 @@ public PartnerMetadata fetchMetadata(String submissionId) throws SQLException {
         }
     }
 
+    @Override
+    public Set<PartnerMetadata> fetchMetadataForMessageLinking(String submissionId)
+            throws SQLException {
+        var sql =
+                """
+                SELECT m2.*
+                FROM metadata m1
+                JOIN metadata m2
+                    ON m1.placer_order_number = m2.placer_order_number
+                        AND m1.sending_application_id = m2.sending_application_id
+                        AND m1.sending_facility_id = m2.sending_facility_id
+                WHERE m1.sent_message_id = ?;
+                """;
+
+        try (Connection conn = connectionPool.getConnection();
+                PreparedStatement statement = conn.prepareStatement(sql)) {
+            statement.setString(1, submissionId);
+            ResultSet resultSet = statement.executeQuery();
+
+            Set<PartnerMetadata> metadataSet = new HashSet<>();
+
+            while (resultSet.next()) {
+                metadataSet.add(partnerMetadataFromResultSet(resultSet));
+            }
+
+            return metadataSet;
+        }
+    }
+
+    @Override
+    public Set<PartnerMetadata> fetchLinkedMessages(String messageId) throws SQLException {
+        var sql =
+                """
```

Comment:
Do we need a nested query here? I would think we can just do a 1 level query to get back the results we need.

Reply:
I could use a self-join instead, or do you have another idea?

Reply:
Changed to use join instead